sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +183,79 @@ private static String namespaceOf(ClassForDisplay clazz) {
-     */
-    ItemBuilder add(String key, Duration value);
-
-    /**
-     * Register the given duration display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, Duration)
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable Duration value);
-
-    /**
-     * Register the given duration display data if the value is different than the specified
-     * default.
-     *
-     * @see DisplayData.Builder#add(String, Duration)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable Duration value, @Nullable Duration defaultValue);
-
-    /**
-     * Register the given class display data. The metadata item will be registered with type
-     * {@link DisplayData.Type#JAVA_CLASS}, and is identified by the specified key and namespace
-     * from the current transform or component.
-     */
-    ItemBuilder add(String key, Class<?> value);
-
-    /**
-     * Register the given class display data. The metadata item will be registered with type
-     * {@link DisplayData.Type#JAVA_CLASS}, and is identified by the specified key and namespace
-     * from the current transform or component.
-     */
-    ItemBuilder add(String key, ClassForDisplay value);
-
-    /**
-     * Register the given class display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable Class<?> value);
-
-    /**
-     * Register the given class display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
+     * Register the given display item if the value is not null.
      */
-    ItemBuilder addIfNotNull(String key, @Nullable ClassForDisplay value);
+    Builder addIfNotNull(Item<?> item);
 
     /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable Class<?> value, @Nullable Class<?> defaultValue);
-
-    /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
+     * Register the given display item if the value is different than the specified default.
      */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable ClassForDisplay value, @Nullable ClassForDisplay defaultValue);
-  /**
-   * Register the given display data with the specified type.
-   *
-   * <p> The added display data is identified by the specified key and namespace from the current
-   * transform or component.
-   *
-   * @throws ClassCastException if the value cannot be safely cast to the specified type.
-   *
-   * @see DisplayData#inferType(Object)
-   */
-    ItemBuilder add(String key, Type type, Object value);
+    <T> Builder addIfNotDefault(Item<T> item, @Nullable T defaultValue);
```

Comment:
It transforms both values and compares them. This is useful particularly for working with similar java types which map to the same display data type, such as `Integer` and `Long`:

    // compares correctly and doesn't add default value.
    builder.addIfNotDefault(
        DisplayData.item("foobar", Long.valueOf(100L)), 
        Integer.valueOf(100));